R/derive_param_diam_to_sev.R

```diff
@@ -0,0 +1,224 @@
+#'   "XYZ1001", "REDNESS", 1.8, "1.8", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1001", "REDNESS", 1.4, "1.4", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1002", "REDNESS", 11.1, "11.1", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "REDNESS", 7.4, "7.4", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "REDNESS", 6, "6", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "REDNESS", 2.1, "2.1", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "REDNESS", 1.1, "1.1", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1001", "SWELLING", 5.5, "5.5", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1001", "SWELLING", 2.5, "2.5", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1001", "SWELLING", 2, "2", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1001", "SWELLING", 1.8, "1.8", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1001", "SWELLING", 1.4, "1.4", "VACCINATION 1", "Diameter", "DIAMETER",
+#'   "XYZ1002", "SWELLING", 10.1, "10.1", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "SWELLING", 7.1, "7.1", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "SWELLING", 5, "5", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "SWELLING", 1.8, "1.8", "VACCINATION 2", "Diameter", "DIAMETER",
+#'   "XYZ1002", "SWELLING", 1.4, "1.4", "VACCINATION 2", "Diameter", "DIAMETER"
+#' )
+#'
+#' derive_param_diam_to_sev(
+#'   dataset = input,
+#'   filter_faobj = c("REDENSS", "SWELLING"),
+#'   filter_diam = "DIAMETER",
+#'   testcd_sev = "SEV",
+#'   test_sev = "Severity"
+#' )
+#'
+#' @export
+#' @keywords der_adxx
+#' @family der_adxx
+#'
+derive_param_diam_to_sev <- function(dataset = NULL,
+                                     filter_diam = "DIAMETER",
+                                     filter_faobj = c("REDNESS", "SWELLING"),
+                                     testcd_sev = "SEV",
+                                     test_sev = "Severity/Intensity",
+                                     none = c(0, 2),
+                                     mild = c(2, 5),
+                                     mod = c(5, 10),
+                                     sev = 10) {
+  #-----------------------------------------------------------------------------
+  # assertion checks
+  #-----------------------------------------------------------------------------
+  assert_data_frame(dataset,
+    required_vars = vars(
+      USUBJID, AVAL, AVALC,
+      FAOBJ, FATEST, FATESTCD
+    )
+  )
+  assert_numeric_vector(arg = c(none, mild, mod, sev), optional = FALSE)
+  assert_character_vector(
+    arg = c(
+      filter_diam, filter_faobj, testcd_sev,
+      test_sev
+    ),
+    optional = FALSE
+  )
+  #----------------------------------------------------------------------------
+  # Checking & Removing the records which has severity records for the FAOBJ
+  #-----------------------------------------------------------------------------
```

Comment:
```suggestion
  # Checking & Removing the records which has severity records for the FAOBJ
```

admiral standards are just to have single comment line - these follow tidyverse standards